Dockerfile

```diff
@@ -3,14 +3,15 @@ FROM jenkins:1.642.4
 USER root
 
 RUN echo "deb http://archive.debian.org/debian jessie main" > /etc/apt/sources.list \
-    && rm /etc/apt/sources.list.d/jessie-backports.list
+    && rm /etc/apt/sources.list.d/jessie-backports.list \
+    && sed -i 's;http://archive.debian.org/debian/;http://deb.debian.org/debian/;' /etc/apt/sources.list
 
 RUN apt-get update \
-    && apt-get install -y apt-transport-https ca-certificates \
-    && apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D \
-    && echo deb https://apt.dockerproject.org/repo debian-jessie main >> /etc/apt/sources.list.d/docker.list \
+    && apt-get install --force-yes -y apt-transport-https ca-certificates curl gnupg-agent software-properties-common sudo \
+    && curl -fsSL https://download.docker.com/linux/debian/gpg | sudo apt-key add - \
```

Comment:
this one initially a remak fro @mbaechler : you are not checking anymore the key you receive

Reply:
it is in the documentation, anyway, if there is an attack to be done here, they should get into the docker infrastructure first, during the 15 minutes the docker image building image last, which is unlikely.
On the other hand, we would have to spend a huge amount of time debugging it when we will build the image in few months.